packages/syft/src/syft/service/enclave/enclave_service.py

```diff
@@ -0,0 +1,209 @@
+        name="send_user_code_inputs_to_enclave",
+        roles=GUEST_ROLE_LEVEL,
+    )
+    def send_user_code_inputs_to_enclave(
+        self,
+        context: AuthedServiceContext,
+        user_code_id: UID,
+        inputs: Dict,
+        node_name: str,
+    ) -> Result[Ok, Err]:
+        if not context.node or not context.node.signing_key:
+            return Err(f"{type(context)} has no node")
+
+        user_code_service = context.node.get_service("usercodeservice")
+        action_service = context.node.get_service("actionservice")
+        user_code = user_code_service.stash.get_by_uid(
+            context.node.signing_key.verify_key, uid=user_code_id
+        )
+        if user_code.is_err():
+            return SyftError(
+                message=f"Unable to find {user_code_id} in {type(user_code_service)}"
+            )
+        user_code = user_code.ok()
+
+        res = user_code.status.mutate(
+            value=UserCodeStatus.EXECUTE,
+            node_name=node_name,
+            verify_key=context.credentials,
+        )
+        if res.is_err():
+            return res
+        user_code.status = res.ok()
+
+        root_context = AuthedServiceContext(credentials=context.node.verify_key)
+        user_code_update = user_code_service.update_code_state(
+            context=root_context, code_item=user_code
+        )
+        if isinstance(user_code_update, SyftError):
+            return user_code_update
+
+        if not action_service.exists(context=context, obj_id=user_code_id):
+            dict_object = DictObject(id=user_code_id)
+            dict_object.base_dict[str(context.credentials)] = inputs
+            action_service.store.set(
+                uid=user_code_id,
+                credentials=context.node.verify_key,
+                syft_object=dict_object,
+                has_result_read_permission=True,
+            )
+
+        else:
+            res = action_service.store.get(
+                uid=user_code_id, credentials=context.node.verify_key
+            )
+            if res.is_ok():
+                dict_object = res.ok()
+                dict_object.base_dict[str(context.credentials)] = inputs
+                action_service.store.set(
+                    uid=user_code_id,
+                    credentials=context.node.verify_key,
+                    syft_object=dict_object,
+                )
+            else:
+                return res
+
+        return Ok(Ok(True))
+
+
+# Checks if the given user code would  propogate value to enclave on acceptance
+def check_enclave_transfer(
+    user_code: UserCode, value: UserCodeStatus, context: ChangeContext
+):
+    if (
+        type(user_code.enclave_metadata).__name__ == "OblvMetadata"
+        and value == UserCodeStatus.EXECUTE
+    ):
+        # relative
+        from ...external import OBLV
+
+        if OBLV:
+            # relative
+            from ...external.oblv.oblv_service import OblvService
```

Comment:
Maybe use `enable_external_lib` to validate oblv package, similar to `packages/syft/src/syft/__init__.py`

Reply:
I have taken  a note of this in the tech debt ticket(https://github.com/OpenMined/Heartbeat/issues/518), since we are going to refactor the whole oblivious interface to use generic interface, maybe we could address this in that ticket.